unit_tests/src/test_ticket_administration.cpp

```diff
@@ -0,0 +1,369 @@
+#include <catch2/catch.hpp>
+
+#include "irods/connection_pool.hpp"
+#include "irods/rodsClient.h"
+#include "irods/ticket_administration.hpp"
+#include "irods/query_builder.hpp"
+#include "irods/user_administration.hpp"
+
+#include <fmt/format.h>
+
+#include <string>
+#include <iostream>
+#include <stdio.h>
+#include <stdexcept>
+
+TEST_CASE("ticket administration")
+{
+    namespace tic = irods::experimental::administration::ticket;
+    std::string read_ticket_name = "testTicket";
+    std::string write_ticket_name = "testTicket2";
+    std::string read_string = "read";
+    std::string read_coll_string = "/tempZone/home/rods";
+    std::string write_string = "write";
+    std::string write_coll_string = "/tempZone";
+
+    irods::experimental::query_builder builder;
+
+    load_client_api_plugins();
+    auto conn_pool = irods::make_connection_pool();
+    auto conn = conn_pool->get_connection();
+
+    std::string set_ticket_name = "setTicket";
+    std::string set_ticket_collection = "/tempZone/home";
+    std::string user_name = "testUser";
+    std::string group_name = "testGroup";
+
+    int set_test_number = 19;
+    std::string set_test_number_string = std::to_string(set_test_number);
+    std::string remove_test_number_string = std::to_string(0);
+
+    irods::experimental::administration::user test_user{user_name};
+    irods::experimental::administration::client::add_user(conn, test_user);
+
+    irods::experimental::administration::group test_group{group_name};
+    irods::experimental::administration::client::add_group(conn, test_group);
+
+    SECTION("create read ticket")
+    {
+        tic::client::create_ticket(conn, tic::type::read, read_coll_string, read_ticket_name);
+
+        auto general_query = builder.build<rcComm_t>(conn, "select TICKET_STRING");
+
+        bool test_ticket_exists = false;
+
+        for (auto&& row : general_query) {
+            for (auto values : row) {
+                if (values.compare(read_ticket_name) == 0) {
+                    test_ticket_exists = true;
+                }
+            }
+        }
+
+        REQUIRE(test_ticket_exists);
+
+        std::string query =
+            fmt::format("select TICKET_COLL_NAME, TICKET_TYPE where TICKET_STRING = '{}'", read_ticket_name);
+        general_query = builder.build<rcComm_t>(conn, query);
+
+        REQUIRE(read_coll_string.compare(general_query.begin().capture_results().at(0)) == 0);
```

Comment:
`std::string` and `std::string_view` support `operator==()` so you can rewrite this as:
```cpp
REQUIRE(read_coll_string == general_query.begin().capture_results().at(0));
```